fmn/fasjson_client.py

```diff
@@ -0,0 +1,55 @@
+import requests
+import requests.exceptions
+from requests.compat import urlencode, urljoin
+from requests_gssapi import HTTPSPNEGOAuth
+
+
+class Client(object):
+    """
+    A fasjson client to make very specific requests to fasjson.
+    Necessary because the official fasjson-client library does not support
+    python3.
+    """
+    def __init__(self, url, principal=None):
+        self.url = url
+        self.principal = principal
+
+        gssapi_auth = HTTPSPNEGOAuth(
+            opportunistic_auth=True, mutual_authentication="OPTIONAL"
```

Comment:
In a [python-freeipa ticket](https://github.com/opennode/python-freeipa/issues/46), Crys recommends creating the credentials and attaching the object to the HTTPSNEGOAuth instance. I'm not entirely sure what the benefits are but I've found out it's always safer to follow Crys' advice ;-)